### ISSUE TITLE
Tests fail if main requires other files [javascript]

### ISSUE THREAD

**Issue by fatadel** [se2p/ps-wise-2021-public-tests#11]
For the javascript version, the tests fail if the main file requires other files.
If you separate functionality into several files, which is **allowed** for the free flow style for javascript, the test will simply fail.
This is related to the way how tests are implemented.

**Comment by fatadel** [comment]
@alessiogambi 
Okay, the issue I was encountering is not directly related to the separation of functionality into multiple files.
There is a much bigger **conceptual** error in the tests and the specification.
So, the specification is talking about **exit codes** for the program errors.
However, the tests are checking for **return codes** of the main method.
These are conceptually different things.
Exit codes are emitted using `process.exit()` method, while return codes are emitted with the `return` statement.
And if one implements the program strictly according to the specification (i.e. using `process.exit()`), the tests will simply fail.

**Comment by fatadel** [comment]
And the worst thing is that the tests **don't signal** anyhow what's wrong, they simply terminate **without any output**, and just `make` signalling that the recipe failed with a particular error code.
And it took me several hours trying to figure out why my **working** program is not passing tests.

**Comment by fatadel** [comment]
So, the specification and the tests are very raw, I would say.

**Comment by alessiogambi** [comment]
> For the javascript version, the tests fail if the main file requires other files.
> If you separate functionality into several files, which is **allowed** for the free flow style for javascript, the test will simply fail.
> This is related to the way how tests are implemented.

Hi, I do not understand if the original issue of importing files in minesweeper.js is still open. It seems that the problem is different and related to (#8 ). In that case maybe we can close this issue and work on #8? Otherwise we should discuss this problem into a separate issue

**Comment by fatadel** [comment]
@alessiogambi 
Yes, I think we can close this particular issue.
By the way, I have found another problem related to stdin and the tests. I guess it will also be resolved after #8.
Unfortunately, before that, the javascript tests seem to be pretty useless.

**Comment by fatadel** [comment]
@alessiogambi 
Yes, I think we can close this particular issue.
Regarding the exit codes, you can simply change the specification, that is, replace "exit codes" => "return codes of the main function", and the inconsistency between the tests and the specification will also be resolved.
